src/module/actor/ABFActor.ts

```diff
@@ -53,6 +57,143 @@ export class ABFActor extends Actor {
+    setTimeout(() => {
+      let supShields = this.system[type][`${type}Shields`];
+      let shieldId = supShields[supShields.length - 1]._id;
+      let args = {
+        thisActor: this,
+        newShield: true,
+        shieldId
+      };
+      executeArgsMacro(newShield.name, args);
+    }, 100);
+  }
+
+  applyDamageSupernaturalShield(
+    supShield: any,
+    damage: number,
+    dobleDamage: boolean,
+    type: string,
+    newCombatResult: any
+  ) {
+    setTimeout(() => {
+      const shieldValue = supShield.system.shieldPoints.value;
+      let shieldId: any;
+      if (supShield.id) {
+        shieldId = supShield.id;
+      } else {
+        let supShields = this.system[type][`${type}Shields`];
+        shieldId = supShields[supShields.length - 1]._id;
+        if (shieldId == undefined) {
+          return ui.notifications.warn('Escudo Sobrenatural no encontrado');
+        }
+      }
+      const newShieldPoints = dobleDamage
+        ? shieldValue - damage * 2
+        : shieldValue - damage;
+      if (newShieldPoints > 0) {
+        let updates: any = [
+          { _id: shieldId, ['system.shieldPoints.value']: newShieldPoints }
+        ];
+        Item.updateDocuments(updates, { parent: this });
+      } else {
+        Item.deleteDocuments([shieldId], { parent: this });
+        if (newShieldPoints < 0 && newCombatResult) {
+          const needToRound = (game as Game).settings.get(
+            'animabf',
+            ABFSettingsKeys.ROUND_DAMAGE_IN_MULTIPLES_OF_5
+          );
+          const result = calculateDamage(
+            newCombatResult.attack,
+            0,
+            newCombatResult.at,
+            Math.abs(newShieldPoints),
+            newCombatResult.halvedAbsorption
+          );
+          const breakingDamage = needToRound ? roundTo5Multiples(result) : result;
+          this.applyDamage(breakingDamage);
+        }
+        let args = {
+          thisActor: this,
+          newShield: false,
+          shieldId
+        };
+        executeArgsMacro(supShield.name, args);
+      }
+    }, 100);
+  }
+
+  accumulateDefenses(keepAccumulating: boolean) {
+    const defensesCounter: any = this.getFlag('world', `${this._id}.defensesCounter`) || {
+      value: true,
+      accumulated: 0
+    };
+    const newDefensesCounter = defensesCounter.accumulated + 1;
+    if (keepAccumulating) {
+      this.update({
+        flags: {
+          world: {
+            [`${this._id}.defensesCounter`]: {
+              accumulated: newDefensesCounter,
+              value: true
+            }
+          }
+        }
+      });
+    } else {
+      this.update({
+        flags: {
+          world: {
+            [`${this._id}.defensesCounter`]: { value: false }
+          }
+        }
+      });
+    }
+  }
+
+  resetDefensesCounter() {
+    this.update({
+      flags: {
+        world: {
+          [`${this._id}.defensesCounter`]: { accumulated: 0 }
+        }
+      }
+    });
```

Comment:
Mejor usar `this.unsetFlag(...)`

Reply:
No se usan flags para esta función

Reply:
Mi primer comentario no es bueno, porque se me cruzó un cable y puse `this.unsetFlag(...)` en lugar de `this.setFlag(...)`. Pero sí que se usan flags, se actualizan los `flags.world`, no?

Reply:
Creo que lo mezclaste con la funcion de arriba "resetDefensesCounter".
En esta función se actualiza el system del actor

Reply:
Ah, yo me refiero a esa función, `resetDefensesCounter` creo. Líneas 164 a 170 del archivo en la PR.

Reply:
Disculpa, lo estaba viendo desde VisualStudio y me marcába otra línea. Estoy corrigiendo todas las funciones que usan flags de la forma que indicaste, usando getFlag y setFlag, les quité el ID del actor y las puse en animabf en vez de world